Blocks/Persistence/file.h

```diff
@@ -158,52 +160,40 @@ class FilePersister {
         // While reading the file, record the offset of each record and store it in `offset`.
         IteratorOverFileOfPersistedEntries<ENTRY> cit(fi, 0, 0);
         std::streampos current_offset(0);
-        while (fi.peek() == constants::kDirectiveMarker
-                   ? ProcessDirective(fi)
-                   : cit.ProcessNextEntry([&](const idxts_t& current, const char*) {
-                     CURRENT_ASSERT(current.index == offset.size());
-                     CURRENT_ASSERT(current.index == timestamp.size());
-                     offset.push_back(current_offset);
-                     timestamp.push_back(current.us);
-                     current_offset = fi.tellg();
-                   })) {
+        auto head = std::chrono::microseconds(-1);
+        while (cit.ProcessNextEntry(
+            [&](const idxts_t& current, const char*) {
+              CURRENT_ASSERT(current.index == offset.size());
+              CURRENT_ASSERT(current.index == timestamp.size());
+              if (!(current.us > head)) {
+                CURRENT_THROW(ss::InconsistentTimestampException(head + std::chrono::microseconds(1), current.us));
+              }
+              offset.push_back(current_offset);
+              timestamp.push_back(current.us);
+              current_offset = fi.tellg();
+              head = current.us;
+              head_offset = 0;
+            },
+            [&](const char* key, const char* value) {
+              if (key == constants::kHeadDirective) {
+                const auto us = std::chrono::microseconds(current::FromString<int64_t>(value));
```

Comment:
Perhaps `typedef` this underlying `int64` type, and put it next to the `"%020lld` constant?